python/plugins/processing/algs/gdal/merge.py

```diff
@@ -129,8 +158,17 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-o')
         arguments.append(out)
 
-        for layer in layers:
-            arguments.append(layer.source())
+        # Always write input files to a text file in case there are many of them and the
```

Comment:
Maybe it makes sense to have this piece as generic function/method and reuse it in all places where we need to create file with inputs?

Reply:
Yep, will do